### PR TITLE
Add Duty Calculator section on the import tab.

### PR DESCRIPTION
**Pull request by alexdesi**
### Jira link
https://transformuk.atlassian.net/browse/HOTT-1162

### What?
Add Duty Calculator section on the import tab.

### Why?
To remove (soon) the overview tab.

## Screenshot
Screenshot 2021-12-08 at 15 19 47

### CODE REVIEW

File: app/views/measures/grouped/_xi.html.erb [hunk 1]
```diff
@@ -4,11 +4,18 @@
...
-    caption: 'Customs duties',
-    collection: xi_import_measures.customs_duties.sort_by(&:key),
-    css_id: 'customs_duties' }
-  %>
+
+  <h3 class="govuk-heading-m govuk-!-margin-top-7 govuk-!-margin-bottom-3", id='customs_duties'>
+    Customs duties
+  </h3>
+
+  <div class="govuk-inset-text govuk-inset-text--s nhs-grey-5 explainer">
```
Comment (willfish) [inline app/views/measures/grouped/_xi.html.erb:12]:
```suggestion
  <div class="govuk-inset-text govuk-inset-text--s explainer">
```

File: app/webpacker/src/stylesheets/_customs_duties_explainer.scss [hunk 2]
```diff
@@ -0,0 +1,13 @@
+.nhs-grey-5 {
```
Comment (willfish) [inline app/webpacker/src/stylesheets/_customs_duties_explainer.scss]: NHS?

Comment (willfish) [reply app/webpacker/src/stylesheets/_customs_duties_explainer.scss]: This is the same colour as the feature panel in app/webpacker/src/stylesheets/_feature-panel.scss. Are we creating a new feature panel with the same background colour? Is there any room for consolidating these two?

File: app/views/measures/grouped/_table_customs_duties.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<table class="small-table measures govuk-table"  >
```
Comment (willfish) [inline app/views/measures/grouped/_table_customs_duties.html.erb]: Why can't we reuse the existing table partial?

File: app/views/measures/grouped/_uk.html.erb [hunk 4]
```diff
@@ -15,12 +15,21 @@
...
-  <%= render partial: 'measures/grouped/table', locals: {
-    caption: 'Customs duties',
-    collection: uk_import_measures.customs_duties.sort_by(&:key),
-    css_id: 'customs_duties'
-  } %>
+  <h3 class="govuk-heading-m govuk-!-margin-top-7 govuk-!-margin-bottom-3", id='customs_duties'>
+    Customs duties
+  </h3>
+
+  <div class="govuk-inset-text govuk-inset-text--s nhs-grey-5 explainer">
```
Comment (willfish) [inline app/views/measures/grouped/_uk.html.erb]: As far as I can see the only thing that's different for this version of the table is this div. I think we should follow the existing pattern and maybe put this div in a partial that we conditionally render depending on a provided local assigns

Comment (willfish) [reply app/views/measures/grouped/_uk.html.erb]: Or convert all of the conditional logic where we special snowflake behaviour in the tables for show_stw_text and hide_duty_rate into their own versions of the table. This way is a half-way house where we're only having some tables that are a bit different having their own partials.

File: app/views/measures/grouped/_xi.html.erb (hunk 1 above)
Comment (willfish) [inline app/views/measures/grouped/_xi.html.erb]: We should put this into a partial, really. Lots of duplication here

**Review by willfish — CHANGES_REQUESTED** [review]
This looks good but we need to:

1. Avoid breaking the existing pattern just so we can render a div (use a conditional like we've done for the show_stw_text, hide_duty_rate , etc
2. Move to rendering the the duty calculator link with a partial that takes a declarable
3. Only render the calculator partial when the declarable says we can (`if declarable.calculate_duties?`)
4. Remove references to the NHS and maybe rationalise some of the colours we're using since they seem to be shared with the feature-panel

File: app/webpacker/src/stylesheets/_customs_duties_explainer.scss (hunk 2 above)
Comment (alexdesi) [reply app/webpacker/src/stylesheets/_customs_duties_explainer.scss]: :D yeah, I've copied the CSS in the prototype, I'll see if I can reuse an existing color

File: app/views/measures/grouped/_table_customs_duties.html.erb (hunk 3 above)
Comment (alexdesi) [reply app/views/measures/grouped/_table_customs_duties.html.erb]: The thing is, the existing partial table, do not contain just the table, but the header too.
And we need the customs duties section after the header.

Comment (alexdesi) [reply app/views/measures/grouped/_table_customs_duties.html.erb]: I guess one way to reuse the existing partial is adding the `yield` and passing a block, I'll give a try

File: app/views/measures/grouped/_table.html.erb [hunk 5]
```diff
@@ -2,6 +2,8 @@
   <%= local_assigns[:caption] %>
 </h3>
 
+<%= yield %>
```
Comment (willfish) [inline app/views/measures/grouped/_table.html.erb:5]:
```suggestion
<%= yield :duty_calculator_link %>
```

Comment (willfish) [inline app/views/measures/grouped/_table.html.erb:5]:
```suggestion
<%= content_for :duty_calculator_link %>
```

Comment (willfish) [inline app/views/measures/grouped/_table.html.erb]:
```suggestion
<%= render 'measures/grouped/tariff_duty_calculator_link', declarable_code: declarable.code if local_assigns[:duty_calculator_link] %>
```

Comment (willfish) [reply app/views/measures/grouped/_table.html.erb]: Not sure why we need a different pattern than what we've got currently here

File: app/views/measures/grouped/_uk.html.erb [hunk 6]
```diff
@@ -11,16 +11,18 @@
...
 <% if uk_import_measures.customs_duties.present? %>
-  <%= render partial: 'measures/grouped/table', locals: {
-    caption: 'Customs duties',
-    collection: uk_import_measures.customs_duties.sort_by(&:key),
-    css_id: 'customs_duties'
-  } %>
+  <%= render 'measures/grouped/table',
+              caption: 'Custom duties',
+              collection: uk_import_measures.customs_duties.sort_by(&:key),
+              css_id: 'customs-duties' do %>
```
Comment (willfish) [inline app/views/measures/grouped/_uk.html.erb]:
```suggestion
              css_id: 'customs-duties',
              duty_calculator_link: true,
              declarable: uk_declarable %>
```

File: app/views/measures/grouped/_tariff_duty_calculator_link.html.erb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<div class="govuk-inset-text govuk-inset-text--s feature-panel--shaded explainer">
```
Comment (willfish) [inline app/views/measures/grouped/_tariff_duty_calculator_link.html.erb]: Don't forget to nest these in

```erb
<% if declarable.calculate_duties? %>
...
<% end %>
```

**Comment by alexdesi** [comment]
## Customs Duties calculator
### Desktop:
Screenshot 2021-12-10 at 16 02 40

### Mobile:
Screenshot 2021-12-10 at 16 15 54

File: app/views/measures/grouped/_table.html.erb (hunk 5 above)
Comment (alexdesi) [reply app/views/measures/grouped/_table.html.erb]: changed, as discussed we choose to use local_assigns[:...]  style for consistency.

File: app/views/measures/grouped/_xi.html.erb (hunk 1 above)
Comment (alexdesi) [reply app/views/measures/grouped/_xi.html.erb]: 👍  refactored!

File: app/views/measures/grouped/_tariff_duty_calculator_link.html.erb (hunk 7 above)
Comment (alexdesi) [reply app/views/measures/grouped/_tariff_duty_calculator_link.html.erb]: maybe that check is redundant,
this view is called on this conditional:

`<% if xi_import_measures.customs_duties.present? %> `

File: app/views/measures/grouped/_uk.html.erb [hunk 8]
```diff
@@ -12,16 +12,17 @@
...
-    caption: 'Customs duties',
-    collection: uk_import_measures.customs_duties.sort_by(&:key),
-    css_id: 'customs_duties'
-  } %>
+  <%= render 'measures/grouped/table',
+              caption: 'Customs duties',
+              collection: uk_import_measures.customs_duties.sort_by(&:key),
+              css_id: 'customs_duties',
+              declarable: uk_declarable,
+              show_duty_calculator: true %>
```
Comment (willfish) [inline app/views/measures/grouped/_uk.html.erb]:
```suggestion
              show_duty_calculator: uk_declarable.calculate_duties? %>

```

File: app/views/measures/grouped/_xi.html.erb [hunk 9]
```diff
@@ -4,11 +4,12 @@
...
-    caption: 'Customs duties',
-    collection: xi_import_measures.customs_duties.sort_by(&:key),
-    css_id: 'customs_duties' }
-  %>
+  <%= render 'measures/grouped/table',
+              caption: 'Custom duties',
+              collection: xi_import_measures.customs_duties.sort_by(&:key),
+              css_id: 'customs_duties',
+              declarable: xi_declarable,
+              show_duty_calculator: true %>
```
Comment (willfish) [inline app/views/measures/grouped/_xi.html.erb]:
```suggestion
              show_duty_calculator: xi_declarable.calculate_duties? %>
```

File: app/views/measures/grouped/_table.html.erb [hunk 10]
```diff
@@ -4,6 +4,8 @@
 
 <%= render 'shared/stw_link', declarable: declarable, search: search, anchor: anchor if local_assigns[:show_stw_text]  %>
 
+<%= render 'measures/grouped/tariff_duty_calculator_link', declarable_code: declarable.code if local_assigns[:show_duty_calculator] && local_assigns[:declarable]%>
```
Comment (willfish) [inline app/views/measures/grouped/_table.html.erb]:
```suggestion
<%= render 'measures/grouped/tariff_duty_calculator_link', declarable_code: declarable.code if local_assigns[:show_duty_calculator] %>
```

File: app/webpacker/src/stylesheets/_customs_duties_explainer.scss [hunk 11]
```diff
@@ -0,0 +1,30 @@
+@include govuk-media-query($from: tablet) {
+  .calculator-panel {
+    background-color: $feature-panel-shaded-background-colour;
+
+    border-left: 10px solid #ccc;
+
+    color: #0b0c0c;
```
Comment (alexdesi) [inline app/webpacker/src/stylesheets/_customs_duties_explainer.scss]: I am trying to rationalize these colors, and move it to _colours.scss

**Review by willfish — APPROVED** [review]

**Pull request merged**